src/transformers/integrations.py

```diff
@@ -327,12 +326,11 @@ def dynamic_modules_import_trainable(*args, **kwargs):
         if is_datasets_available():
             import datasets.load
 
-            dynamic_modules_path = os.path.join(datasets.load.init_dynamic_modules(), "__init__.py")
+            from transformers.utils import direct_import
+
+            path = datasets.load.init_dynamic_modules()
             # load dynamic_modules from path
-            spec = importlib.util.spec_from_file_location("datasets_modules", dynamic_modules_path)
-            datasets_modules = importlib.util.module_from_spec(spec)
-            sys.modules[spec.name] = datasets_modules
-            spec.loader.exec_module(datasets_modules)
+            direct_import("datasets_modules", path)
```

Comment:
This wasn't part of the cited [PR](https://github.com/huggingface/transformers/commit/9d1116e9951686f937d17697820117636bfc05a5) that this refactors, but I thought I'd add since it since it's using the same import methods

Reply:
This won't work here as we need the `sys.modules[spec.name] = datasets_module` line that is not in `direct_import` (it does it the other way around because of the specificities of the Transformers inits).